### ISSUE TITLE
Wizard Step 4: Sponsorship UI Feature Upgrade

### ISSUE THREAD

**Issue by mariobehling** [fossasia/open-event-frontend#5546]
The sponsorship form needs a feature upgrade to cater for business users. Please see below pointers how this can work.

There should be two areas of the form:
1. Sponsorship Level
Here the organizers can define sponsorship levels (previously called types) like Platinum, Gold etc.
2. Sponsors and Partners
Here the specific sponsors are added into a level, that was created in step 1 above.
3. Preview
Here we show how the sponsorship area would look like.
---

**1. Sponsorship Level**

Name* [Platinum Sponsor] [Number of logo columns] [+]   ↓ 
Name* [Gold Sponsor]     [Number of logo columns] [+] ↑ ↓
Name* [Silver Sponsor]   [Number of logo columns] [+] ↑ ↓
Name* [Bronze Sponsor]   [Number of logo columns] [+] ↑ 

- when a sponsorship type is created, automatically create a section to add sponsors below
- if an organizer wants to delete a sponsorship type, show a pop up message with this text "If you delete this sponsorship type sponsors that are saved in the level will be deleted. Do you want to proceed? Yes / No"
- if a sponsorship type is moved up or down, I think it is difficult to reflect this change right way and also move the below sections. Therefore show a message on the page itself (not as a pop up) when the user move a section up or down: "Changes to the order of the sponsorship sections will be reflected after saving the event."

--

**2. Sponsors and Partners**

Platinum Sponsor

(left column)
Name [Red Hat]
URL [https://redhatsomething.com]
Description [Such a great company|

(right column)
Top right: "Remove" button / show up down arrow to place sponsors within a sponsorship type, and i a sponsor is already on the top and the user clicks up arrow again move the sponsor to the upper type (or lower type respectively) ↑ ↓

logo

[Add sponsor]
(rename from "add another sponsor" to "Add sponsor")

--

Gold..
(same as above)

--

Silver
(same as above)

--

Bronze
(same as above)

--

**3. Preview**

(Show preview of the sponsorship area here)

**Comment by iamareebjamal** [comment]
This will improve the UI and option in the wizard, but still not solve this issue - https://github.com/fossasia/open-event-frontend/pull/5532#issuecomment-724259354

Secondly, we'll have to discuss where we'll store the column information such that the database changes are not backward incompatible

**Comment by mariobehling** [comment]
Correct, this is why I just opened a separate issue https://github.com/fossasia/open-event-frontend/issues/5547. But it also give the user already more control here if we have the preview area as you suggested.

**Comment by mariobehling** [comment]
Regarding backward compatibility - if this is too much to take care of just find a simple solution and we go into the paid events and fix it in the form manually.

**Comment by iamareebjamal** [comment]
I'm afraid there is no simple solution and the only way this can be achieved is to store the column size with each sponsor like the type and level is being stored. Then it creates a conf;ict if 2 sponsors are in same level/type but have different columns. I was going to go with it already if level and type were synced 1:1, so we could extract it easily but that's not the case and thus same problem happens here. 

In case we do store column size with each sponsor, and ignore that it can cause conflict and when we group the sponsors by level/type, we pick the first column size out of those grouped sponsors. That is the easiest implementation for now.
